lib/bson/decimal128.rb

```diff
@@ -107,11 +107,18 @@ def ==(other)
     #
     # @since 4.2.0
     def initialize(initial, digits = nil)
-      if initial.is_a?(String)
-        set_bits(*Builder::FromString.new(initial).bits)
-      else
-        big_decimal = BigDecimal.new(*[initial, digits].compact)
-        set_bits(*Builder::FromBigDecimal.new(big_decimal).bits)
+      case initial
+        when String
+          set_bits(*Builder::FromString.new(initial).bits)
+        when Integer
+          set_bits(*Builder::FromInteger.new(initial).bits)
+        when Float
+          set_bits(*Builder::FromFloat.new(initial).bits)
```

Comment:
I've added support for specifying digits of precision when passing a Float to Decimal128#new  https://github.com/mongodb/bson-ruby/pull/61/commits/37bd9dde968073e7833a6d529a3ec4fda3a01272